api/src/main/java/io/strimzi/api/kafka/model/balancing/KafkaRebalanceState.java

```diff
@@ -53,23 +55,37 @@
      *     <dt>ProposalReady</dt><dd>If the user sets annotation strimzi.io/rebalance=refresh and the proposal is already ready.</dd>
      * </dl>
      */
-    Stopped,
+    Stopped(List.of(KafkaRebalanceAnnotation.refresh)),
     /**
      * There's been some error.
      * Transitions to:
      * <dl>
      *     <dt>New</dt><dd>If the error was caused by the resource itself that was fixed by the user.</dd>
      * </dl>
      */
-    NotReady,
+    NotReady(List.of(KafkaRebalanceAnnotation.refresh)),
     /**
      * The rebalance is complete and there is no transition from this state.
      * The resource is eligible for garbage collection after a configurable delay.
      * There is no transition from this state to a new one.
      */
-    Ready,
+    Ready(List.of(KafkaRebalanceAnnotation.refresh)),
     /**
      * The user paused reconciliations by setting annotation strimzi.io/pause-reconciliation="true".
      */
-    ReconciliationPaused
+    ReconciliationPaused(List.of());
+
+    private final List<KafkaRebalanceAnnotation> validAnnotations;
+
+    KafkaRebalanceState(List<KafkaRebalanceAnnotation> validAnnotations) {
+        this.validAnnotations = validAnnotations;
+    }
+
+    public boolean validateAnnotation(KafkaRebalanceAnnotation annotation) {
+        return validAnnotations.contains(annotation);
+    }
+
+    public String validAnnotationString() {
```

Comment:
why not overriding the `toString` instead of naming a new method?
Something like:

```java
@Override
    public String toString() {
        return "KafkaRebalanceState{" +
                "validAnnotations=" + validAnnotations +
                '}';
    }
```

Reply:
Hi, I guess we are making use of the normal toString() method at some places so overriding it might cause errors.

Reply:
for eg. -> `KafkaRebalanceState.valueOf(rebalanceStateType)` If Iam not wrong valueOf() does call the `toString` method, right?

Reply:
Not sure about the `valueOf()` but what I know is that `toString()` calls `Enum.name()` by default. Can you try maybe?

Reply:
I did tried to override the `toString()` method but I was getting some errors and  this `KafkaRebalanceState.valueOf(rebalanceStateType)` was in the stacktrace

Reply:
As agreed offline, let's stay as we are without overriding `toString()` but making `validAnnotationString()` a getter for the `validAnnotations` instead.